nbconvert/exporters/slides.py

```diff
@@ -1,28 +1,78 @@
 """HTML slide show Exporter class"""
 
-#-----------------------------------------------------------------------------
-# Copyright (c) 2013, the IPython Development Team.
-#
+# Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
-#
-# The full license is in the file COPYING.txt, distributed with this software.
-#-----------------------------------------------------------------------------
 
-#-----------------------------------------------------------------------------
-# Imports
-#-----------------------------------------------------------------------------
+from copy import deepcopy
 
-from nbconvert import preprocessors
 from traitlets.config import Config
+from traitlets import Unicode
 
 from .html import HTMLExporter
 
-#-----------------------------------------------------------------------------
-# Classes
-#-----------------------------------------------------------------------------
+def prepare(nb):
+    """Add some convenenience metadata on cells for the slide template.
+    """
+    nb = deepcopy(nb)
+
+    for cell in nb.cells:
+        # Make sure every cell has a slide_type
+        cell.metadata.slide_type = cell.metadata.get('slideshow', {}).get('slide_type', '-')
+
+    # Find the first visible cell
+    for index, cell in enumerate(nb.cells):
+        if cell.metadata.slide_type not in {'notes', 'skip'}:
+            cell.metadata.slide_type = 'slide'
```

Comment:
We are forcing the `slide_type=slide` in a cell maybe no labelled as is... but I am OK with that because we are working in a copy of the nb, not the nb itself, so changing the property from what it was selected thought the user interface should not be a problem anymore...

Additionally, what about detecting the slide_type before this so we can warn the user about how properly start an slideshow with the `slyde_type=slide` instead of `-` or `subslide`? I guess we should tell people that we are using this "not slide" first visible cell as slide beginning because they never start the slideshow with the proper slide_type.
I think a warning in the console (but proceed with the conversion, obviously) should be enough in this case...